src/service/video-manager-impl.js

```diff
@@ -902,6 +912,39 @@ class VideoEntry {
 const AUTO_FULLSCREEN_ID_PROP = '__AMP_AUTO_FULLSCREEN_ID__';
 
 
+/**
+ * @param {!AmpElement} video
+ * @return {boolean}
+ * @restricted
+ */
+function canFullScreen(win, video) {
+  // Safari and iOS can only fullscreen <video> elements directly. In cases
+  // where the player component is implemented via an iframe, we need to rely
+  // on a postMessage API to fullscreen. Such an API is not necessarily provided
+  // by all players.
+  const platform = Services.platformFor(win);
+  if (!(platform.isIos() || platform.isSafari())) {
+    return true;
+  }
+  if (supportsFullscreenViaApi(video)) {
+    return true;
+  }
+  const internalElement = getInternalElementFor(video);
+  return internalElement.tagName.toLowerCase() == 'video';
```

Comment:
I suggest running this as the first check to short-circuit for the common amp-video case.

Reply:
Sure, done.